cleanlab/outlier.py

```diff
@@ -0,0 +1,541 @@
+          For details, see `features` in :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+        pred_probs : np.ndarray, optional
+          An array of shape ``(N, K)`` of model-predicted probabilities.
+          For details, see `pred_probs` in :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+        Returns
+        -------
+        scores : np.ndarray
+          If `features` are passed in, `ood_features_scores` are returned.
+          The score is based on the average distance between the example and its K nearest neighbors in the dataset
+          (in feature space).
+
+          If `pred_probs` are passed in, `ood_predictions_scores` are returned.
+          The score is based on the model predicted probabilities.
+        """
+        self._assert_valid_inputs(features, pred_probs)
+
+        if features is not None:
+            if self.knn is None:
+                raise ValueError(
+                    f"OOD Object needs to be fit on features first. Call fit() or fit_scores() before this function."
+                )
+            else:
+                params = self._get_params(self.OUTLIER_PARAMS)  # get params specific to outliers
+                scores = _get_ood_features_scores(
+                    features, self.knn, **params, return_estimator=False
+                )
+
+        if pred_probs is not None:
+            if self.confident_thresholds is None and self.params["adjust_pred_probs"]:
+                raise ValueError(
+                    f"OOD Object needs to be fit on features first. Call fit() or fit_scores() before this function."
+                )
+            else:
+                params = self._get_params(self.OOD_PARAMS)  # get params specific to outliers
+                scores = _get_ood_predictions_scores(
+                    pred_probs,
+                    confident_thresholds=self.confident_thresholds,
+                    **params,
+                    return_thresholds=False,
+                )
+
+        # TODO: How to fix this typing issue without assert? _get_[...]_scores should always return np.ndarray
+        #  since return_estimator and return_thresholds are False.
+        assert isinstance(scores, np.ndarray)
+        return scores
+
+    def _get_params(self, param_keys) -> dict:
+        """
+        Helper method to get function specific dictionary of parameters (i.e. only those in param_keys).
+        """
+        return {k: v for k, v in self.params.items() if k in param_keys}
+
+    def _get_invalid_params(self, params, param_keys) -> list:
+        """
+        Helper method to get list of parameters in param that are not in param_keys.
+        """
+        if params is None:
+            return []
+        return list(set(params.keys()).difference(set(param_keys)))
+
+    def _assert_valid_inputs(self, features, pred_probs):
```

Comment:
doesn't seem like this should be instance method (does not use `self` AFAICT)

Reply:
```suggestion
    @staticmethod
    def _assert_valid_inputs(features, pred_probs):
```
allows you to use `self._assert_valid_inputs(features, pred_probs)` within the scope of the class without setting `self` as the first argument.

Reply:
Thank you! This is a lot cleaner :D